pkg/apis/messaging/v1alpha1/channel_validation.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"knative.dev/pkg/kmp"
+
+	eventingduck "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/apis"
+)
+
+func (c *Channel) Validate(ctx context.Context) *apis.FieldError {
+	return c.Spec.Validate(ctx).ViaField("spec")
+}
+
+func (cs *ChannelSpec) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+
+	if cs.ChannelTemplate == nil {
+		errs = errs.Also(apis.ErrMissingField("channelTemplate"))
```

Comment:
Add a comment that users aren't expected to set this. The Channel defaulter is expected to do so.

Reply:
Done, thanks!